pkg/apis/rke.cattle.io/v1/rkecontrolplane_types.go

```diff
@@ -0,0 +1,180 @@
+
+	// ETCDSnapshotRestorePhase is the current phase the etcd snapshot
+	// restore operation is currently executing.
+	// +kubebuilder:validation:Enum=Started;Shutdown;Restore;PostRestorePodCleanup;InitialRestartCluster;PostRestoreNodeCleanup;RestartCluster;Finished;Failed
+	// +optional
+	ETCDSnapshotRestorePhase ETCDSnapshotPhase `json:"etcdSnapshotRestorePhase,omitempty"`
+
+	// ETCDSnapshotCreate is the state for which the last etcd snapshot
+	// create operation was successful.
+	// +optional
+	ETCDSnapshotCreate *ETCDSnapshotCreate `json:"etcdSnapshotCreate,omitempty"`
+
+	// ETCDSnapshotCreatePhase is the current phase the etcd snapshot create
+	// operation is currently executing.
+	// +kubebuilder:validation:Enum=Started;RestartCluster;Finished;Failed
+	// +optional
+	ETCDSnapshotCreatePhase ETCDSnapshotPhase `json:"etcdSnapshotCreatePhase,omitempty"`
+
+	// ConfigGeneration is the current generation of the configuration for a
+	// given cluster.
+	// Changing this value (which is done automatically during an etcd restore) will trigger a reconciliation loop
+	// which will invoke draining (if enabled).
+	// +optional
+	ConfigGeneration int64 `json:"configGeneration,omitempty"`
+
+	// Initialized denotes that the API server is initialized and worker
+	// nodes can be joined to the cluster.
+	// +optional
+	Initialized bool `json:"initialized,omitempty"`
+
+	// AgentConnected denotes that the cluster-agent connection is currently
+	// established for the cluster.
+	// +optional
+	AgentConnected bool `json:"agentConnected,omitempty"`
+}
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=rkecontrolplanes,shortName=rcp,scope=Namespaced,categories=cluster-api
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels="cluster.x-k8s.io/v1beta1=v1"
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
+// +kubebuilder:printcolumn:name="Initialized",type=string,JSONPath=".status.initialized",description="This denotes whether or not the control plane is initialized"
+// +kubebuilder:printcolumn:name="API Server Available",type=boolean,JSONPath=".status.ready",description="RKEControlPlane API Server is ready to receive requests"
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=".metadata.creationTimestamp",description="Time duration since creation of RKEControlPlane"
+// +kubebuilder:printcolumn:name="Version",type=string,JSONPath=".spec.kubernetesVersion",description="Kubernetes version associated with this control plane"
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RKEControlPlane is the Schema for the controlplane.
+type RKEControlPlane struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
```

Comment:
This doesn't seem to have any effect, either the description or the optional marker. I suspect kubebuilder always generates metadata in the same way, and kubectl explain always prints the same.